plugins/search-indexer-backend/src/types.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Base properties that all indexed documents must include, as well as some
+ * common properties that documents are encouraged to use where appropriate.
+ */
+export interface IndexableDocument {
+  /**
+   * The primary name of the document (e.g. name, title, identifier, etc).
+   */
+  title: string;
```

Comment:
I've been looking at this one for a little bit now and don't know if I prefer having the required one as "id" instead as it in 8 times out of 10 probably will be a combination of name namespace and kind for all software catalog entities or something similar and for example for TechDocs where title actually exist its more clear to extend with title instead of extend with id? or..? Not sure if its a big deal but couldn't stop think of it haha!

Reply:
Thanks for bringing this up!  Very much worth discussing since, whether we collapse a catalog entity's name and a TechDocs document's title into a single attribute, or make them separate attributes, could impact our ability to make a quality default "global" search experience.

I was thinking about how, internally, everything that isn't a catalog entity becomes something of a second-class citizen in search because the field containing an entity's main identifier is weighted so high by default, and other types of results (like a TechDocs page) don't really use that field in a meaningful way.  Perhaps we'd get to better search results easier if we had a single field that semantically meant "the main way this thing is referred to."

Reply:
That's a very great reasoning, thanks for brining that perspective. As we have been discussed before, we do not want to tie it to the software catalog to much be able to make sure that things outside of the catalog is treated the same way. So I like it, lets keep it!